webapp/src/main/resources/application.sql.yml

```diff
@@ -10,14 +10,19 @@ sql:
       where natural_id in (:naturalIds)
 
   group:
+    findAllDistinctSchoolYears: >-
+      select distinct sg.school_year
+      from student_group sg
+        join school sc on sc.id = sg.school_id
+      where 1=:statewide or sc.district_id in (:district_ids) or sg.school_id in (:school_ids)
+
     findAllByQuery: >-
       select sg.name as group_name,
         sc.name as school_name,
-        su.code as subject_code,
+        sg.subject_id,
         sg.school_year
       from student_group sg
         join school sc on sc.id = sg.school_id
-        left join subject su on su.id = sg.subject_id
```

Comment:
Since we don't care about denorm as much for subject_codes, I can remove this join and use the id which maps to the enum.